source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -290,12 +316,21 @@ void Wavm::link(absl::string_view name) {
     emscriptenInstance_ = Emscripten::instantiate(compartment_, irModule_);
     rootResolver.moduleNameToInstanceMap().set("env", emscriptenInstance_->env);
     rootResolver.moduleNameToInstanceMap().set("asm2wasm", emscriptenInstance_->asm2wasm);
-    rootResolver.moduleNameToInstanceMap().set("global", emscriptenInstance_->global);
   }
   WAVM::Runtime::LinkResult linkResult = linkModule(irModule_, rootResolver);
   moduleInstance_ = instantiateModule(compartment_, module_, std::move(linkResult.resolvedImports),
                                       std::string(name));
   memory_ = getDefaultMemory(moduleInstance_);
+  for (auto& p : intrinsicGlobals_) {
```

Comment:
Feel free to ignore: This green area is quite isolated from the existing logic. It doesn't use any stack variable so it's reasonable to name this piece some other day.